apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -123,6 +124,11 @@ type OpenTelemetryCollectorSpec struct {
 	//
 	// +optional
 	Autoscaler *AutoscalerSpec `json:"autoscaler,omitempty"`
+	// PodDisruptionBudget specifies the pod disruption budget configuration to use
+	// for the OpenTelemetryCollector workload.
+	//
+	// +optional
+	PodDisruptionBudget *PodDisruptionBudgetSpec `json:"podDisruptionBudget,omitempty"`
```

Comment:
note for other reviewers, this is a paired down version of the main PDB spec because we don't want to allow the configuration of the selector